app/v2/services/UpscanService.scala

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v2.services
+
+import cats.data.EitherT
+import com.google.inject.ImplementedBy
+import com.google.inject.Inject
+import config.AppConfig
+import uk.gov.hmrc.http.HeaderCarrier
+import v2.connectors.UpscanConnector
+import v2.models.errors.UpscanInitiateError
+import v2.models.request.UpscanInitiate
+import v2.models.responses.UpscanInitiateResponse
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.util.control.NonFatal
+
+@ImplementedBy(classOf[UpscanServiceImpl])
+trait UpscanService {
+
+  def upscanInitiate()(implicit
+    headerCarrier: HeaderCarrier,
+    executionContext: ExecutionContext
+  ): EitherT[Future, UpscanInitiateError, UpscanInitiateResponse]
+
+}
+
+class UpscanServiceImpl @Inject() (
+  upscanConnector: UpscanConnector,
+  appConfig: AppConfig
+) extends UpscanService {
+
+  override def upscanInitiate()(implicit
+    headerCarrier: HeaderCarrier,
+    executionContext: ExecutionContext
+  ): EitherT[Future, UpscanInitiateError, UpscanInitiateResponse] = {
+
+    val upscanInitiate = UpscanInitiate("https://myservice.com/callback", None, None, None, None) // TODO callback url will be configured later
```

Comment:
@drnaylor-oc I think we should be setting the max file size that we accept?

Reply:
Actually, yes, that's a good idea. It should be configurable, but it should be set to 22MB for now (we may drop it to 20MB to give us headroom)